pkg/services/secrets/kvstore/migrations/migrator.go

```diff
@@ -39,36 +42,53 @@ func ProvideSecretMigrationProvider(
 	migrateToPluginService *MigrateToPluginService,
 	migrateFromPluginService *MigrateFromPluginService,
 ) *SecretMigrationProviderImpl {
-	services := make([]SecretMigrationService, 0)
-	services = append(services, dataSourceSecretMigrationService)
+	migServices := make([]SecretMigrationService, 0)
+	migServices = append(migServices, dataSourceSecretMigrationService)
 	// Plugin migration should always be last; should either migrate to or from, not both
 	// This is because the migrateTo checks for use_plugin = true, in which case we should always
 	// migrate by default to ensure users don't lose access to secrets. If migration has
 	// already occurred, the migrateTo function will be called but it won't do anything
 	if cfg.SectionWithEnvOverrides("secrets").Key("migrate_from_plugin").MustBool(false) {
-		services = append(services, migrateFromPluginService)
+		migServices = append(migServices, migrateFromPluginService)
 	} else {
-		services = append(services, migrateToPluginService)
+		migServices = append(migServices, migrateToPluginService)
 	}
 
-	return &SecretMigrationProviderImpl{
+	s := &SecretMigrationProviderImpl{
 		ServerLockService:        serverLockService,
-		services:                 services,
+		migServices:              migServices,
 		migrateToPluginService:   migrateToPluginService,
 		migrateFromPluginService: migrateFromPluginService,
+		errs:                     make(chan error),
 	}
+
+	s.BasicService = services.NewBasicService(s.start, s.running, nil).WithName(modules.SecretMigrator)
+	return s
 }
 
-func (s *SecretMigrationProviderImpl) Run(ctx context.Context) error {
+func (s *SecretMigrationProviderImpl) start(ctx context.Context) error {
 	return s.Migrate(ctx)
 }
 
+func (s *SecretMigrationProviderImpl) running(ctx context.Context) error {
+	select {
+	case err, ok := <-s.errs:
```

Comment:
This is wrong; I'm not sure what makes sense for a `running` func here. Failures in the sub-services don't kill this service 🤔